### ISSUE TITLE
new Item interface is misbehaving

### ISSUE THREAD

**Issue by williamconifer** [getk2/k2-v3-dev-build#296]
I have a brand new K2 Ver 3 downloaded and installed today sitting on Joomla 3.5.1 running RocketTheme Intersteller theme which is Gantry 5. No other plugins installed. PHP is 5.3.3

In backend the New or Edit Item page loads awkwardly with side "scroll wrapper" left menu infringing in the layout and the top right spinner doesn't stop. 

Developer Tools console is showing the following: 
"index.php?option=com_k2:300 Uncaught TypeError: Cannot read property 'initialize' of undefined"

I have attached a screenshot.

Because the content area is partially covered I have to click on revisions tab which then kicks me out of the edit window and back to the Item list. I click again on an item and i then get to the edit page. The typeerror still occurs but I can see and edit text. I can save the item. 

HOWEVER, when I go to K2 Image tab I click on the Upload button, I am able to select an image from my local machine and I click open. BUT nothing happens in the edit form. No upload progress, no image. I save the item and the image is not there. The image does not appear in the Media folder. 

I have to do a full on evaluation for future use for a potential client and this is really throwing me off. 

Help
jack

![tdg_sandbox_april_2016_-_administration_-_k2](https://cloud.githubusercontent.com/assets/6101622/14398369/d4bff61c-fda0-11e5-8681-ab45ec3abbb7.jpg)

**Comment by wbarnes** [comment]
The recommended minimum requirements for Gantry 5:
http://docs.gantry.org/gantry5/basics/system-requirements

php5.4

^potential to be your bug.

Regards,
William

**Comment by williamconifer** [comment]
Actually that was a typo on my part. This site is using PHP 5.5.33. I've also use PHP 5.6 and 7 with the same issues.

**Comment by paulvanharen** [comment]
Dear @williamconifer 

What you describe are issues on the back-end of your installation. I expect that the frontend theme has no influence on the back-end behavior. I justed loaded the latest k2 v3 on a clean J3.5.1 installation and I could not reproduce your findings. 

However, the phenomena you've described are often caused by JS errors in the back-end. The implementation of text editors, such as Tiny MCE and JCE in Joomla has many variations, and is known to give rise to JS errors in k2 v3. Per today, I have good results with Tiny MCE and JCE; I did not manage to get other editors (Rockpad) up and running yet. 

Would you be so kind to check for the editor you're using and switch back to Tiny MCE to verify if the problem persists?

Thanks, Paul

**Comment by williamconifer** [comment]
Hey Paul,

Please note that the problem I was having was with the TinyMCE set as the editor. 

When I installed JCE it all worked just fine. 

If I set Joomla default editor to NONE. I get this warning displaying when I go to Items or Categories in K2:  Warning[2] DOMDocument::loadHTML(): htmlParseEntityRef: expecting ';' in Entity, line: 50. Line 90 in file /var/www/vhosts/domainname.com/httpdocs/administrator/components/com_k2/classes/editor.php

I see you folks worked on administrator/components/com_k2/classes/editor.php a couple of weeks ago, but it does not look like the "dirty fix" is working all that good. 

I hope this helps.

**Comment by kavadas** [comment]
@williamconifer 

The error you see in your console indicates that is raised from an inline script. So the page source at line 300.

**Comment by paulvanharen** [comment]
@williamconifer

I have reimplemented parts of the editor start-up code. TinyMCE, JCE and RokPad should now work as expected. Would you be so kind to verify if the problem you had is now gone?

Thanks and regards, Paul

**Comment by williamconifer** [comment]
Hey Paul, thanks for following up. Should I reinstall K2 v3 on this site or is it better to just use specific file(s) that you have updated? I'm not as git'fluent as I should be.

**Comment by paulvanharen** [comment]
Hi Jack, the easiest is to download the ZIP from Github (in the green button "Clone or download") and install/update this as any regular Joomla extension. That should do the job.

Best regards, Paul

**Comment by williamconifer** [comment]
OK I have tested on a sandbox of Joomla 3.6.2 running K2 ver 3 (from today). The display issue I was experiencing with the scroll wrapper has been fixed. I checked with TinyMCE and JCE.

However the K2 Item image upload process does not work. This is how I described it back in April:

"HOWEVER, when I go to K2 Image tab I click on the Upload button, I am able to select an image from my local machine and I click open. BUT nothing happens in the edit form. No upload progress, no image. I save the item and the image is not there. The image does not appear in the Media folder."
